hack/istio/multicluster/kiali-prepare-remote-cluster.sh

```diff
@@ -245,6 +245,13 @@ create_kiali_remote_cluster_secret() {
     local cert_auth_yaml="certificate-authority-data: ${remote_cluster_ca_bytes}"
   fi
 
+  # Check if the REMOTE_CLUSTER_NAME matches the AWS EKS ARN regex pattern.
+  # If it does, extract the cluster name from the ARN and assign it back to REMOTE_CLUSTER_NAME.
+  AWS_REGION_REGEX='(us|eu|ap|sa|ca|me)-(north|south|east|west|central)-\d'
+  AWS_EKS_ARN_REGEX="^arn:aws:eks:${AWS_REGION_REGEX}:\d{12}:cluster\/[A-Za-z0-9]+(?:-[A-Za-z0-9]+)*$"
+  if echo "${REMOTE_CLUSTER_NAME}" | grep -Eq "${AWS_EKS_ARN_REGEX}"; then
+      REMOTE_CLUSTER_NAME=$(echo "${REMOTE_CLUSTER_NAME}" | sed 's/^.*:cluster\/\(.*\)$/\1/')
+  fi
```

Comment:
This works. You can get rid of the two commented lines that start with "If you want to use sed rather than BASH_REMATCH" -- use what you feel is best. That would work as well as using BASH_REMATCH. I just wanted to provided the two ways I found that can be used to do it (BASH_REMATCH or using sed).

```suggestion
  AWS_REGEX='^arn:aws:eks:(us|eu|ap|sa|ca|me)-(north|south|east|west|central)-[0-9]:[0-9]{12}:cluster/([A-Za-z0-9\-]+)$'
  if [[ ${REMOTE_CLUSTER_NAME} =~ ${AWS_REGEX} ]]; then
      REMOTE_CLUSTER_NAME=${BASH_REMATCH[3]}
      # If you want to use sed rather than BASH_REMATCH:
      # REMOTE_CLUSTER_NAME=$(echo "${REMOTE_CLUSTER_NAME}" | sed 's/^.*:cluster\/\(.*\)$/\1/')
  fi
```